Added another Advanced solution to problem Intermediate Algorithm Scripting: Wherefore art thou. Fixed a link not listed as the others in the page. (#36287)
freeCodeCamp/freeCodeCamp: guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/wherefore-art-thou/index.md

*   <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for' target='_blank' rel='nofollow'>For Loops</a>
*   <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter' target='_blank' rel='nofollow'>Array.prototype.filter()</a>
*   <a href='https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty' target='_blank' rel='nofollow'>Object.hasOwnProperty()</a>

> *	<a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys' target='_blank' rel='nofollow'>Object.keys()</a>

Reviewer: ```suggestion
* <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys' target='_blank' rel='nofollow'>Object.keys()</a>
```

> *   <a>Array.prototype.reduce()</a>

Reviewer: Add links to these too if possible. not required

*   <a href='https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty' target='_blank' rel='nofollow'>Object.hasOwnProperty()</a>

## ![:rotating_light:](https://forum.freecodecamp.com/images/emoji/emoji_one/rotating_light.png?v=3 ":rotating_light:") Advanced Code Solution:

>     function whatIsInAName(collection, source) {

Reviewer: ```suggestion
function whatIsInAName(collection, source) {
```

>         return collection.filter(a =>

Reviewer: ```suggestion
  return collection.filter(a =>
```

>     		Object.keys(source).every(key => a[key] === source[key])

Reviewer: ```suggestion
    Object.keys(source).every(key => a[key] === source[key])
```

Reviewer 2: Please use variable names which describe the content instead of something generic like `a` which conveys no meaning.

Author: Will do

> 			);

Reviewer: ```suggestion
  );
```

>     }

Reviewer: ```suggestion
}
```

>     whatIsInAName([{ first: "Romeo", last: "Montague" }, { first: "Mercutio", last: null }, { first: "Tybalt", last: "Capulet" }], { last: "Capulet" });

Reviewer: ```suggestion
whatIsInAName([{ first: "Romeo", last: "Montague" }, { first: "Mercutio", last: null }, { first: "Tybalt", last: "Capulet" }], { last: "Capulet" });
```
Also, add 3 backticks `\`` after this line

> ![:rocket:](https://forum.freecodecamp.com/images/emoji/emoji_one/rocket.png?v=3 ":rocket:") <a href='https://repl.it/CLmj/0' target='_blank' rel='nofollow'>Run Code</a>

Reviewer: ```suggestion
```
We're not accepting non-fCC controlled links anymore since the user of the website might remove them

### Code Explanation:

> *   Because the resulting array should be a subset of `collection`, we use `Array.filter`

Reviewer: ```suggestion
* Because the resulting array needs to be a subset of `collection`, we use `Array.filter`
```

> *   Following this, the condition is only to return an array that contains objects whose key and key value are the same as in `source`, or in other words; for every key value pair `source` has, check if `collection` has any objects with the same key value pair.

Reviewer: ```suggestion
*  Following this, the condition is only to return an array that contains objects whose key and key value are the same as in `source`, or in other words; for every key-value pair `source` has, check if `collection` has any objects with the same key value pair.
```

> *   Using `Array.every` function we can quickly check that for every key value pair in `source` there is an object in `collection` that has the same key value pairs. 

Reviewer: ```suggestion
* Using the `Array.every` function we can quickly check that for every key-value pair in `source` there is an object in `collection` that has the same key value pairs. 
```

> *   This condition is sufficient to fulfill the problem condition

Reviewer: ```suggestion
* This condition is sufficient to fulfill the problem condition
```

#### Relevant Links

> *   <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter' target='_blank' rel='nofollow'>Array.prototype.filter()</a>

Reviewer: ```suggestion
* <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter' target='_blank' rel='nofollow'>Array.prototype.filter()</a>
```

> *   <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every' target='_blank' rel='nofollow'>Array.prototype.every()</a>

Reviewer: ```suggestion
* <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every' target='_blank' rel='nofollow'>Array.prototype.every()</a>
```

> *	<a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys' target='_blank' rel='nofollow'>Object.keys()</a>

Reviewer: ```suggestion
* <a href= 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys' target='_blank' rel='nofollow'>Object.keys()</a>
```
